In [1]:
import pandas as pd
import numpy as np


In [2]:
df = pd.read_csv('data.csv')

In [3]:
df.head()

title                author  \
0      Climate action needs new frontline leadership  Ozawa Bineshi Albert   
1        The dark history of the overthrow of Hawaii         Sydney Iaukea   
2     How play can spark new ideas for your business         Martin Reeves   
3  Why is China appointing judges to combat clima...     James K. Thornton   
4     Cement's carbon problem — and 2 ways to fix it       Mahendra Singhi   

             date   views  likes  \
0   December 2021  404000  12000   
1   February 2022  214000   6400   
2  September 2021  412000  12000   
3    October 2021  427000  12000   
4    October 2021    2400     72   

                                                link  
0  https://ted.com/talks/ozawa_bineshi_albert_cli...  
1  https://ted.com/talks/sydney_iaukea_the_dark_h...  
2  https://ted.com/talks/martin_reeves_how_play_c...  
3  https://ted.com/talks/james_k_thornton_why_is_...  
4  https://ted.com/talks/mahendra_singhi_cement_s...

In [4]:
df.tail()

title            author           date  \
5435  The best stats you've ever seen      Hans Rosling  February 2006   
5436      Do schools kill creativity?  Sir Ken Robinson  February 2006   
5437              Greening the ghetto     Majora Carter  February 2006   
5438                 Simplicity sells       David Pogue  February 2006   
5439      Averting the climate crisis           Al Gore  February 2006   

         views    likes                                               link  
5435  15000000   458000  https://ted.com/talks/hans_rosling_the_best_st...  
5436  72000000  2100000  https://ted.com/talks/sir_ken_robinson_do_scho...  
5437   2900000    88000  https://ted.com/talks/majora_carter_greening_t...  
5438   2000000    60000  https://ted.com/talks/david_pogue_simplicity_s...  
5439   3600000   109000  https://ted.com/talks/al_gore_averting_the_cli...

In [5]:
df

title                author  \
0         Climate action needs new frontline leadership  Ozawa Bineshi Albert   
1           The dark history of the overthrow of Hawaii         Sydney Iaukea   
2        How play can spark new ideas for your business         Martin Reeves   
3     Why is China appointing judges to combat clima...     James K. Thornton   
4        Cement's carbon problem — and 2 ways to fix it       Mahendra Singhi   
...                                                 ...                   ...   
5435                    The best stats you've ever seen          Hans Rosling   
5436                        Do schools kill creativity?      Sir Ken Robinson   
5437                                Greening the ghetto         Majora Carter   
5438                                   Simplicity sells           David Pogue   
5439                        Averting the climate crisis               Al Gore   

                date     views    likes  \
0      December 2021    404000    12000   
1      February 2022    214000     6400   
2     September 2021    412000    12000   
3       October 2021    427000    12000   
4       October 2021      2400       72   
...              ...       ...      ...   
5435   February 2006  15000000   458000   
5436   February 2006  72000000  2100000   
5437   February 2006   2900000    88000   
5438   February 2006   2000000    60000   
5439   February 2006   3600000   109000   

                                                   link  
0     https://ted.com/talks/ozawa_bineshi_albert_cli...  
1     https://ted.com/talks/sydney_iaukea_the_dark_h...  
2     https://ted.com/talks/martin_reeves_how_play_c...  
3     https://ted.com/talks/james_k_thornton_why_is_...  
4     https://ted.com/talks/mahendra_singhi_cement_s...  
...                                                 ...  
5435  https://ted.com/talks/hans_rosling_the_best_st...  
5436  https://ted.com/talks/sir_ken_robinson_do_scho...  
5437  https://ted.com/talks/majora_carter_greening_t...  
5438  https://ted.com/talks/david_pogue_simplicity_s...  
5439  https://ted.com/talks/al_gore_averting_the_cli...  

[5440 rows x 6 columns]

In [6]:
df.isnull().sum()

title     0
author    1
date      0
views     0
likes     0
link      0
dtype: int64

In [7]:
df = df.dropna()

In [8]:
df

title                author  \
0         Climate action needs new frontline leadership  Ozawa Bineshi Albert   
1           The dark history of the overthrow of Hawaii         Sydney Iaukea   
2        How play can spark new ideas for your business         Martin Reeves   
3     Why is China appointing judges to combat clima...     James K. Thornton   
4        Cement's carbon problem — and 2 ways to fix it       Mahendra Singhi   
...                                                 ...                   ...   
5435                    The best stats you've ever seen          Hans Rosling   
5436                        Do schools kill creativity?      Sir Ken Robinson   
5437                                Greening the ghetto         Majora Carter   
5438                                   Simplicity sells           David Pogue   
5439                        Averting the climate crisis               Al Gore   

                date     views    likes  \
0      December 2021    404000    12000   
1      February 2022    214000     6400   
2     September 2021    412000    12000   
3       October 2021    427000    12000   
4       October 2021      2400       72   
...              ...       ...      ...   
5435   February 2006  15000000   458000   
5436   February 2006  72000000  2100000   
5437   February 2006   2900000    88000   
5438   February 2006   2000000    60000   
5439   February 2006   3600000   109000   

                                                   link  
0     https://ted.com/talks/ozawa_bineshi_albert_cli...  
1     https://ted.com/talks/sydney_iaukea_the_dark_h...  
2     https://ted.com/talks/martin_reeves_how_play_c...  
3     https://ted.com/talks/james_k_thornton_why_is_...  
4     https://ted.com/talks/mahendra_singhi_cement_s...  
...                                                 ...  
5435  https://ted.com/talks/hans_rosling_the_best_st...  
5436  https://ted.com/talks/sir_ken_robinson_do_scho...  
5437  https://ted.com/talks/majora_carter_greening_t...  
5438  https://ted.com/talks/david_pogue_simplicity_s...  
5439  https://ted.com/talks/al_gore_averting_the_cli...  

[5439 rows x 6 columns]

In [9]:
df.values

array([['Climate action needs new frontline leadership',
        'Ozawa Bineshi Albert', 'December 2021', 404000, 12000,
        'https://ted.com/talks/ozawa_bineshi_albert_climate_action_needs_new_frontline_leadership'],
       ['The dark history of the overthrow of Hawaii', 'Sydney Iaukea',
        'February 2022', 214000, 6400,
        'https://ted.com/talks/sydney_iaukea_the_dark_history_of_the_overthrow_of_hawaii'],
       ['How play can spark new ideas for your business',
        'Martin Reeves', 'September 2021', 412000, 12000,
        'https://ted.com/talks/martin_reeves_how_play_can_spark_new_ideas_for_your_business'],
       ...,
       ['Greening the ghetto', 'Majora Carter', 'February 2006', 2900000,
        88000, 'https://ted.com/talks/majora_carter_greening_the_ghetto'],
       ['Simplicity sells', 'David Pogue', 'February 2006', 2000000,
        60000, 'https://ted.com/talks/david_pogue_simplicity_sells'],
       ['Averting the climate crisis', 'Al Gore', 'February 2006

In [10]:
data = df.values

In [11]:
data[0]

array(['Climate action needs new frontline leadership',
       'Ozawa Bineshi Albert', 'December 2021', 404000, 12000,
       'https://ted.com/talks/ozawa_bineshi_albert_climate_action_needs_new_frontline_leadership'],
      dtype=object)

In [12]:
data[5435]

array(['Do schools kill creativity?', 'Sir Ken Robinson', 'February 2006',
       72000000, 2100000,
       'https://ted.com/talks/sir_ken_robinson_do_schools_kill_creativity'],
      dtype=object)

In [13]:
data[5435][0]

'Do schools kill creativity?'

In [14]:
data[5435][1]

'Sir Ken Robinson'

In [15]:
author = []

for i in range(len(data)):
    author.append(data[i][1])

In [16]:
author

['Ozawa Bineshi Albert',
 'Sydney Iaukea',
 'Martin Reeves',
 'James K. Thornton',
 'Mahendra Singhi',
 'Rosamund Adoo-Kissi-Debrah',
 'Iseult Gillespie',
 'Gay Gordon-Byrne',
 'Erin Meezan',
 'Dwan Reece',
 'Ian Kerner',
 'Richard Browning',
 'Daniel H. Pink',
 'Lucas Joppa',
 ' Belle and Sebastian',
 'Mounia Akl',
 'Yana Buhrer Tavanier',
 'Susan Ruffo',
 'Tim Dunn',
 ' Let It Happen',
 'Dave Matthews',
 'Zoë Karl-Waithaka',
 'Dan Kwartler',
 'Dan Jørgensen',
 'Lei Li',
 'Nick Turner and Whitney Pennington Rodgers',
 'Elizabeth Cox',
 ' Kaz',
 'Katie McGinty',
 'Rachel Kyte',
 'Katie Paterson',
 'Emily Zobel Marshall',
 'Deval Patrick',
 'Rha Goddess and Deepa Purushothaman',
 'Zac Goldsmith',
 'Thomas W. Cronin',
 'Gala Marija Vrbanic',
 'Li-Huei Tsai',
 'Emily Pilloton-Lam',
 'Dustin Burke',
 'Marie Brodsky',
 'Gabrielle Walker',
 'Reuben Jonathan Miller',
 'Iseult Gillespie',
 'Carlos M. Duarte',
 'Candace Parker',
 'Jenny Scheinman with Nels Cline and Todd Sickafoose',
 'Emma Bry

Total Number of Authors who have given a talk

In [17]:
len(author)

5439

In [18]:
distinct_author = set(author)

In [19]:
distinct_author

{'Aaron Sams',
 'Gowri Shankar',
 'Sophie Andrews',
 'Deborah Gordon',
 'OluTimehin Adegbeye',
 'Netta Schramm',
 'Andres Lozano',
 'Dexter Dias',
 'Tabetha Boyajian',
 'Eric Topol',
 'Itamar Mann',
 'Raspyni Brothers',
 'Negin Farsad',
 'Eric Mead',
 'Ole Scheeren',
 'Chiara Decaroli',
 'Asaf Bar-Yosef',
 'Sivamani',
 'Giles Duley',
 'Colombe Cahen-Salvador',
 'Tara Djokic',
 'Bernie Krause',
 'Tyrone Davis, Jr.',
 'Adong Judith',
 'Aaron Reedy',
 'Kaitlyn Sadtler and Heather J. Faust',
 'Becky  Blanton',
 'Dorothy Roberts',
 'Gabby Rivera',
 'François Candelon',
 'Ndidi Nwuneli',
 'Keith Weed',
 'Sophie Hunger',
 'Carole Cadwalladr',
 'Sascha Morrell',
 'Michael Mieni',
 'Kelli Anderson',
 'Anna Rosling Rönnlund',
 'Tina Seelig',
 'Phil Hansen',
 'Leo Igwe',
 'Tomer Garzberg',
 'Mike Cannon-Brookes',
 'Carmen Jiménez Borja',
 'Philippa Neave',
 'Heather Brooke',
 'Danny Dorling',
 'Shawn Achor',
 'Mac Stone',
 'Antionette Carroll',
 'Kate Marvel',
 'Paco de Leon',
 'Anne Basting',
 '

In [20]:
len(distinct_author)

4443

Total Number of Authors who have given a talk is 4443

TED talks given by Bill Gates

In [21]:
for i in range(len(data)):
    if(data[i][1] == 'Bill Gates'):
         print(data[i]) 

['The innovations we need to avoid a climate disaster' 'Bill Gates'
 'March 2021' 1700000 53000
 'https://ted.com/talks/bill_gates_the_innovations_we_need_to_avoid_a_climate_disaster']
['How the pandemic will shape the near future' 'Bill Gates' 'June 2020'
 4600000 138000
 'https://ted.com/talks/bill_gates_how_the_pandemic_will_shape_the_near_future']
['How we must respond to the coronavirus pandemic' 'Bill Gates'
 'March 2020' 8600000 259000
 'https://ted.com/talks/bill_gates_how_we_must_respond_to_the_coronavirus_pandemic']
["The next outbreak? We're not ready" 'Bill Gates' 'March 2015' 43000000
 1300000
 'https://ted.com/talks/bill_gates_the_next_outbreak_we_re_not_ready']
['Teachers need real feedback' 'Bill Gates' 'May 2013' 4500000 135000
 'https://ted.com/talks/bill_gates_teachers_need_real_feedback']
['How state budgets are breaking US schools' 'Bill Gates' 'March 2011'
 2100000 63000
 'https://ted.com/talks/bill_gates_how_state_budgets_are_breaking_us_schools']
['Innovating to

In [22]:
df

title                author  \
0         Climate action needs new frontline leadership  Ozawa Bineshi Albert   
1           The dark history of the overthrow of Hawaii         Sydney Iaukea   
2        How play can spark new ideas for your business         Martin Reeves   
3     Why is China appointing judges to combat clima...     James K. Thornton   
4        Cement's carbon problem — and 2 ways to fix it       Mahendra Singhi   
...                                                 ...                   ...   
5435                    The best stats you've ever seen          Hans Rosling   
5436                        Do schools kill creativity?      Sir Ken Robinson   
5437                                Greening the ghetto         Majora Carter   
5438                                   Simplicity sells           David Pogue   
5439                        Averting the climate crisis               Al Gore   

                date     views    likes  \
0      December 2021    404000    12000   
1      February 2022    214000     6400   
2     September 2021    412000    12000   
3       October 2021    427000    12000   
4       October 2021      2400       72   
...              ...       ...      ...   
5435   February 2006  15000000   458000   
5436   February 2006  72000000  2100000   
5437   February 2006   2900000    88000   
5438   February 2006   2000000    60000   
5439   February 2006   3600000   109000   

                                                   link  
0     https://ted.com/talks/ozawa_bineshi_albert_cli...  
1     https://ted.com/talks/sydney_iaukea_the_dark_h...  
2     https://ted.com/talks/martin_reeves_how_play_c...  
3     https://ted.com/talks/james_k_thornton_why_is_...  
4     https://ted.com/talks/mahendra_singhi_cement_s...  
...                                                 ...  
5435  https://ted.com/talks/hans_rosling_the_best_st...  
5436  https://ted.com/talks/sir_ken_robinson_do_scho...  
5437  https://ted.com/talks/majora_carter_greening_t...  
5438  https://ted.com/talks/david_pogue_simplicity_s...  
5439  https://ted.com/talks/al_gore_averting_the_cli...  

[5439 rows x 6 columns]

Total Number of views of all the TED talks

In [23]:
views = []

for i in range(len(data)):
    views.append(data[i][3])

In [24]:
sum(views)


11214972679

Average view per TED talk

In [25]:
average = int(sum(views)/len(data))

In [26]:
average

2061954

Average number of views for Bill Gates

In [27]:
bill_views = []

for i in range(len(data)):
    if(data[i][1] == 'Bill Gates'):
         bill_views.append(data[i][3])

In [28]:
Average =int(sum(bill_views)/len(bill_views))
Average

9725000

View to Like ratio of the TED talk

In [29]:
view_to_like = []

for i in range(len(data)):
    view_to_like.append(int(data[i][3]/data[i][4]))

In [30]:
view_to_like

[33,
 33,
 34,
 35,
 33,
 35,
 34,
 35,
 34,
 34,
 33,
 33,
 34,
 35,
 33,
 34,
 35,
 34,
 33,
 33,
 32,
 33,
 35,
 35,
 33,
 33,
 36,
 34,
 33,
 34,
 34,
 34,
 33,
 33,
 32,
 33,
 33,
 33,
 34,
 33,
 33,
 34,
 34,
 34,
 34,
 31,
 33,
 32,
 31,
 32,
 33,
 31,
 32,
 34,
 33,
 34,
 33,
 33,
 34,
 34,
 34,
 32,
 33,
 34,
 33,
 33,
 34,
 33,
 32,
 33,
 33,
 33,
 34,
 33,
 35,
 31,
 31,
 33,
 32,
 33,
 33,
 33,
 32,
 31,
 32,
 33,
 33,
 32,
 33,
 32,
 33,
 31,
 33,
 31,
 33,
 33,
 34,
 33,
 31,
 31,
 31,
 31,
 33,
 31,
 35,
 33,
 33,
 33,
 35,
 33,
 33,
 33,
 31,
 33,
 33,
 33,
 33,
 33,
 32,
 35,
 32,
 33,
 34,
 33,
 32,
 34,
 32,
 33,
 33,
 33,
 33,
 34,
 33,
 33,
 31,
 31,
 33,
 33,
 31,
 35,
 33,
 31,
 33,
 33,
 33,
 33,
 33,
 33,
 33,
 33,
 31,
 34,
 35,
 33,
 33,
 33,
 33,
 33,
 31,
 32,
 33,
 33,
 31,
 31,
 32,
 33,
 33,
 33,
 32,
 33,
 32,
 32,
 33,
 34,
 32,
 33,
 33,
 34,
 33,
 33,
 33,
 32,
 32,
 31,
 32,
 34,
 33,
 33,
 33,
 31,
 33,
 32,
 35,
 33,
 32,
 32,
 34,
 32,
 32,
 31,


What is the maximum view to like ratio?

In [31]:
max(view_to_like)

36

What is the minimum view to like ratio?

In [32]:
min(view_to_like)

30

In [33]:
df['vtl_ratio'] = view_to_like

C:\Users\HP\anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [34]:
df

title                author  \
0         Climate action needs new frontline leadership  Ozawa Bineshi Albert   
1           The dark history of the overthrow of Hawaii         Sydney Iaukea   
2        How play can spark new ideas for your business         Martin Reeves   
3     Why is China appointing judges to combat clima...     James K. Thornton   
4        Cement's carbon problem — and 2 ways to fix it       Mahendra Singhi   
...                                                 ...                   ...   
5435                    The best stats you've ever seen          Hans Rosling   
5436                        Do schools kill creativity?      Sir Ken Robinson   
5437                                Greening the ghetto         Majora Carter   
5438                                   Simplicity sells           David Pogue   
5439                        Averting the climate crisis               Al Gore   

                date     views    likes  \
0      December 2021    404000    12000   
1      February 2022    214000     6400   
2     September 2021    412000    12000   
3       October 2021    427000    12000   
4       October 2021      2400       72   
...              ...       ...      ...   
5435   February 2006  15000000   458000   
5436   February 2006  72000000  2100000   
5437   February 2006   2900000    88000   
5438   February 2006   2000000    60000   
5439   February 2006   3600000   109000   

                                                   link  vtl_ratio  
0     https://ted.com/talks/ozawa_bineshi_albert_cli...         33  
1     https://ted.com/talks/sydney_iaukea_the_dark_h...         33  
2     https://ted.com/talks/martin_reeves_how_play_c...         34  
3     https://ted.com/talks/james_k_thornton_why_is_...         35  
4     https://ted.com/talks/mahendra_singhi_cement_s...         33  
...                                                 ...        ...  
5435  https://ted.com/talks/hans_rosling_the_best_st...         32  
5436  https://ted.com/talks/sir_ken_robinson_do_scho...         34  
5437  https://ted.com/talks/majora_carter_greening_t...         32  
5438  https://ted.com/talks/david_pogue_simplicity_s...         33  
5439  https://ted.com/talks/al_gore_averting_the_cli...         33  

[5439 rows x 7 columns]

Finding TED Talks of Bill Gates by short method. 

In [35]:
df[df['author'] == 'Bill Gates']

title      author  \
403   The innovations we need to avoid a climate dis...  Bill Gates   
789         How the pandemic will shape the near future  Bill Gates   
952     How we must respond to the coronavirus pandemic  Bill Gates   
3251                 The next outbreak? We're not ready  Bill Gates   
3878                        Teachers need real feedback  Bill Gates   
4560          How state budgets are breaking US schools  Bill Gates   
4824                                Innovating to zero!  Bill Gates   
5062                   Mosquitos, malaria and education  Bill Gates   

               date     views    likes  \
403      March 2021   1700000    53000   
789       June 2020   4600000   138000   
952      March 2020   8600000   259000   
3251     March 2015  43000000  1300000   
3878       May 2013   4500000   135000   
4560     March 2011   2100000    63000   
4824  February 2010   8100000   245000   
5062  February 2009   5200000   156000   

                                                   link  vtl_ratio  
403   https://ted.com/talks/bill_gates_the_innovatio...         32  
789   https://ted.com/talks/bill_gates_how_the_pande...         33  
952   https://ted.com/talks/bill_gates_how_we_must_r...         33  
3251  https://ted.com/talks/bill_gates_the_next_outb...         33  
3878  https://ted.com/talks/bill_gates_teachers_need...         33  
4560  https://ted.com/talks/bill_gates_how_state_bud...         33  
4824  https://ted.com/talks/bill_gates_innovating_to...         33  
5062  https://ted.com/talks/bill_gates_mosquitos_mal...         33

In [36]:
int(df[df['author'] == 'Bill Gates']['views'].mean())

9725000

In [37]:
df['author'].describe()

count             5439
unique            4443
top       Alex Gendler
freq                45
Name: author, dtype: object

In [38]:
df[df['author'] == 'Alex Gendler']

title        author  \
66    Blood, concrete, and dynamite: Building the Ho...  Alex Gendler   
329                     The woman who stared at the sun  Alex Gendler   
348     How one design flaw almost toppled a skyscraper  Alex Gendler   
358   Demolition, disease, and death: Building the P...  Alex Gendler   
367        How the world's tallest skyscraper was built  Alex Gendler   
372      Why are airplanes slower than they used to be?  Alex Gendler   
522   Building the world's largest (and most controv...  Alex Gendler   
525              Can you solve the monster duel riddle?  Alex Gendler   
548       Can you solve the Alice in Wonderland riddle?  Alex Gendler   
757            The Egyptian myth of the death of Osiris  Alex Gendler   
827                    Which voting system is the best?  Alex Gendler   
864                            How do ventilators work?  Alex Gendler   
943   How the world's longest underwater tunnel was ...  Alex Gendler   
988                Can you solve the death race riddle?  Alex Gendler   
990   The Egyptian myth of Isis and the seven scorpions  Alex Gendler   
1026    Epic Engineering: Building the Brooklyn Bridge   Alex Gendler   
1041  Everything changed when the fire crystal got s...  Alex Gendler   
1093          Can you solve the dragon jousting riddle?  Alex Gendler   
1137   Why doesn't the Leaning Tower of Pisa fall over?  Alex Gendler   
1146             Can you outsmart this logical fallacy?  Alex Gendler   
1165            The myth of Loki and the master builder  Alex Gendler   
1285             Can you solve the secret sauce riddle?  Alex Gendler   
1289                           A brief history of chess  Alex Gendler   
1317             A day in the life of a Cossack warrior  Alex Gendler   
1441    Why should you read "The Master and Margarita"?  Alex Gendler   
1480        Why should you read "Crime and Punishment"?  Alex Gendler   
1700      Can you solve the multiplying rabbits riddle?  Alex Gendler   
1763              Can you solve the giant iron riddle?   Alex Gendler   
1788                               The myth of Sisyphus  Alex Gendler   
1871                               The myth of Hercules  Alex Gendler   
1910                         How does impeachment work?  Alex Gendler   
1923           Can you solve the rebel supplies riddle?  Alex Gendler   
2597                   Can you solve the pirate riddle?  Alex Gendler   
2664               Can you solve the three gods riddle?  Alex Gendler   
2990                     History vs. Napoleon Bonaparte  Alex Gendler   
3107             Can you solve the prisoner hat riddle?  Alex Gendler   
3146                   Can you solve the bridge riddle?  Alex Gendler   
3193  Can you solve the famously difficult green-eye...  Alex Gendler   
3220             The wars that inspired Game of Thrones  Alex Gendler   
3325                            What is a gift economy?  Alex Gendler   
3387                         Why elephants never forget  Alex Gendler   
3416                   History vs. Christopher Columbus  Alex Gendler   
3559                                  How tsunamis work  Alex Gendler   
3610            Why do we cry? The three types of tears  Alex Gendler   
3812           Myths and misconceptions about evolution  Alex Gendler   

                date     views   likes  \
66     December 2021    724000   21000   
329         May 2021   1900000   57000   
348         May 2021    712000   21000   
358       April 2021    724000   21000   
367       April 2021    770000   23000   
372       April 2021   2200000   66000   
522    December 2020    810000   24000   
525    December 2020   1800000   55000   
548    November 2020   1600000   48000   
757        July 2020   2500000   77000   
827        June 2020    856000   25000   
864         May 2020   2300000   70000   
943       March 2020   2800000   85000   
988       March 2020   1400000   42000   
990       March 2020   2000000   60000   
1026   February 202

In [39]:
df[df['author'] == 'Alex Gendler']['views'].sum()

187196000

Finding the talks which were related to Space

In [40]:
topic = 'space'

for i in range(len(data)):
    if(topic in data[i][0].lower()):
        print(data[i])


['How film captures the space between hope and despair' 'Mounia Akl'
 'December 2021' 628000 18000
 'https://ted.com/talks/mounia_akl_how_film_captures_the_space_between_hope_and_despair']
['Yes, scientists are actually building an elevator to space'
 'Fabio Pacucci' 'December 2021' 714000 21000
 'https://ted.com/talks/fabio_pacucci_yes_scientists_are_actually_building_an_elevator_to_space']
['Documentary films that explore trauma — and make space for healing'
 'Almudena Toral' 'May 2021' 1700000 52000
 'https://ted.com/talks/almudena_toral_documentary_films_that_explore_trauma_and_make_space_for_healing']
['What should humans take to space (and leave behind)?'
 'Jorge Mañes Rubio' 'May 2021' 1500000 47000
 'https://ted.com/talks/jorge_manes_rubio_what_should_humans_take_to_space_and_leave_behind']
['Playful, wondrous public spaces built for community and possibility'
 'Matthew Mazzotta' 'May 2021' 1200000 37000
 'https://ted.com/talks/matthew_mazzotta_playful_wondrous_public_spaces_bu

Finding the talks which were related to Black hole.

In [41]:
topic = 'black hole'

for i in range(len(data)):
    if(topic in data[i][0].lower()):
        print(data[i])


['Could we harness the power of a black hole?' 'Fabio Pacucci'
 'October 2020' 464000 13000
 'https://ted.com/talks/fabio_pacucci_could_we_harness_the_power_of_a_black_hole']
["Hawking's black hole paradox explained" 'Fabio Pacucci' 'October 2019'
 3100000 95000
 'https://ted.com/talks/fabio_pacucci_hawking_s_black_hole_paradox_explained']
['The most detailed map of galaxies, black holes and stars ever made'
 'Juna Kollmeier' 'April 2019' 2600000 80000
 'https://ted.com/talks/juna_kollmeier_the_most_detailed_map_of_galaxies_black_holes_and_stars_ever_made']
['Can a black hole be destroyed?' 'Fabio Pacucci' 'May 2019' 1700000 51000
 'https://ted.com/talks/fabio_pacucci_can_a_black_hole_be_destroyed']
['Inside the black hole image that made history' 'Sheperd Doeleman'
 'April 2019' 3600000 108000
 'https://ted.com/talks/sheperd_doeleman_inside_the_black_hole_image_that_made_history']
['Could the Earth be swallowed by a black hole?' 'Fabio Pacucci'
 'September 2018' 1100000 35000
 'https:

Finding the talks which were related to Disaster.

In [42]:
topic = 'disaster'

for i in range(len(data)):
    if(topic in data[i][0].lower()):
        print(data[i])

['The innovations we need to avoid a climate disaster' 'Bill Gates'
 'March 2021' 1700000 53000
 'https://ted.com/talks/bill_gates_the_innovations_we_need_to_avoid_a_climate_disaster']
['An ingenious solution for aid in disaster zones' 'A. Dara Dotz'
 'October 2018' 8600 259
 'https://ted.com/talks/a_dara_dotz_an_ingenious_solution_for_aid_in_disaster_zones']
['These robots come to the rescue after a disaster' 'Robin Murphy'
 'May 2015' 1100000 35000
 'https://ted.com/talks/robin_murphy_these_robots_come_to_the_rescue_after_a_disaster']
['A new mission for veterans — disaster relief' 'Jake Wood'
 'December 2011' 585000 17000
 'https://ted.com/talks/jake_wood_a_new_mission_for_veterans_disaster_relief']
['How to step up in the face of disaster' "Caitria + Morgan O'Neill"
 'June 2012' 976000 29000
 'https://ted.com/talks/caitria_morgan_o_neill_how_to_step_up_in_the_face_of_disaster']
['Cheap, effective shelter for disaster relief' 'Michael McDaniel'
 'February 2012' 318000 9500
 'https:/

Finding the talks which were related to India.

In [43]:
topic = 'india'

for i in range(len(data)):
    if(topic in data[i][0].lower()):
        print(data[i])

['Why was India split into two countries?' 'Haimanti Roy' 'June 2021'
 1700000 51000
 'https://ted.com/talks/haimanti_roy_why_was_india_split_into_two_countries']
["How India could pull off the world's most ambitious energy transition"
 'Varun Sivaram' 'October 2020' 1500000 45000
 'https://ted.com/talks/varun_sivaram_how_india_could_pull_off_the_world_s_most_ambitious_energy_transition']
['How the coronavirus is impacting India — and what needs to happen next'
 'Gayathri Vasudevan' 'April 2020' 251000 7500
 'https://ted.com/talks/gayathri_vasudevan_how_the_coronavirus_is_impacting_india_and_what_needs_to_happen_next']
['5 steps for clean air in India' 'Arunabha Ghosh' 'May 2019' 261000 7800
 'https://ted.com/talks/arunabha_ghosh_5_steps_for_clean_air_in_india']
["How India's local recyclers could solve plastic pollution"
 'Mani Vajipey' 'May 2019' 284000 8500
 'https://ted.com/talks/mani_vajipey_how_india_s_local_recyclers_could_solve_plastic_pollution']
['A bold plan to empower 1.6 m

Finding the talks which were related to Covid.

In [44]:
topic = 'covid'

for i in range(len(data)):
    if(topic in data[i][0].lower()):
        print(data[i])

['4 ways the COVID-19 pandemic changed the way we sleep' 'Matt Walker'
 'December 2021' 295000 8800
 'https://ted.com/talks/matt_walker_4_ways_the_covid_19_pandemic_changed_the_way_we_sleep']
['How the COVID-19 vaccines were created so quickly'
 'Kaitlyn Sadtler and Elizabeth Wayne' 'August 2021' 1300000 41000
 'https://ted.com/talks/kaitlyn_sadtler_and_elizabeth_wayne_how_the_covid_19_vaccines_were_created_so_quickly']
['How COVID-19 reshaped US cities' 'Kevin J. Krizek' 'March 2021' 1500000
 46000
 'https://ted.com/talks/kevin_j_krizek_how_covid_19_reshaped_us_cities']
['How NASA invented a ventilator for COVID-19 ... in 37 days' 'Dan Goods'
 'December 2020' 1200000 38000
 'https://ted.com/talks/dan_goods_how_nasa_invented_a_ventilator_for_covid_19_in_37_days']
['How COVID-19 transformed the future of medicine' 'Daniel Kraft'
 'September 2020' 1300000 40000
 'https://ted.com/talks/daniel_kraft_how_covid_19_transformed_the_future_of_medicine']
['The global cooperation that accelerated